onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperator.cpp

```diff
@@ -635,6 +635,40 @@ namespace Dml
             ));
     }
 
+    void DmlOperator::ConvertNHWCToNCHW(
+        const uint32_t& dimCount, 
+        const gsl::span<const uint32_t> nhwcSizes,
+        std::vector<uint32_t>& nchwSizes,
+        std::vector<uint32_t>& nchwInputStrides)
+    {
+        int i = 0;
+        const uint32_t inputBatch = nhwcSizes[ i++ ];
+        const uint32_t inputDepth = dimCount == 5 ? nhwcSizes[ i++ ] : 0;
+        const uint32_t inputHeight = nhwcSizes[ i++ ];
+        const uint32_t inputWidth = nhwcSizes[ i++ ];
+        const uint32_t inputChannels = nhwcSizes[ i++ ];
```

Comment:
>nhwcSizes[ i++ ] [](http://example.com/codeflow?start=39&length=16)

There really should be an `assert` at a minimum if there are size assumptions about the incoming span.

Reply:
Adding 
```
        assert(dimCount == 5 || dimCount == 4)
```
in next commit

Reply:
Working on this one, need to refactor for any dim count